### PR TITLE
update llvm-dis header with available options

### LINKED ISSUE #108069: [llvm-dis][docs] Update the list of options on the file header
The current llvm-dis file mention only the help command on the header, should be updated with current available options.

https://github.com/llvm/llvm-project/blob/02c943a7ea7b798026baf34f108c18a5cde3821a/llvm/tools/llvm-dis/llvm-dis.cpp#L8-L16

Current options

https://github.com/llvm/llvm-project/blob/02c943a7ea7b798026baf34f108c18a5cde3821a/llvm/tools/llvm-dis/llvm-dis.cpp#L39-L85

- Make sure to not add hidden options

### PR DESCRIPTION
**Pull request by txff99**
update llvm-dis header options

Closes #108069

### CODE REVIEW

File: llvm/tools/llvm-dis/llvm-dis.cpp [hunk 1]
```diff
@@ -12,7 +12,12 @@
 //                            to the x.ll file.
 //  Options:
 //      --help   - Output information about command line switches
-//
+//      -o <filename>                  - Override output filename
+//      -f                             - Enable binary output on terminals
+//      --disable-output               - Don't output the .ll file
+//      --show-annotations             - Add informational comments to the .ll file
+//      --materialize-metadata         - Load module without materializing metadata, then materialize only the metadata
+//
```
Comment (AmrDeveloper) [inline llvm/tools/llvm-dis/llvm-dis.cpp:20]: Check llvm-dis -help to get all options

```
OPTIONS:

Color Options:

  --color                - Use colors in output (default=autodetect)

Disassembler Options:

  -f                     - Enable binary output on terminals
  --materialize-metadata - Load module without materializing metadata, then materialize only the metadata
  -o <filename>          - Override output filename
  --show-annotations     - Add informational comments to the .ll file

Generic Options:

  --help                 - Display available options (--help-hidden for more)
  --help-list            - Display list of available options (--help-list-hidden for more)
  --version              - Display the version of this program
  ```

Comment (txff99) [reply llvm/tools/llvm-dis/llvm-dis.cpp]: got it. I just added all the options in `llvm-dis -help`

**Comment by marcauberer** [comment]
@txff99 Please fix formatting. Otherwise lgtm

**Comment by txff99** [comment]
@marcauberer format fixed! ready to be merged now

**Review by marcauberer — APPROVED** [review]

**Pull request merged**